infra/pr_helper.py

```diff
@@ -0,0 +1,224 @@
+#!/usr/bin/env python
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Adds comments for PR to provide more information for approvers."""
+import base64
+import json
+import os
+import subprocess
+
+import requests
+import yaml
+
+OWNER = 'google'
+REPO = 'oss-fuzz'
+GITHUB_URL = 'https://github.com/'
+API_URL = 'https://api.github.com'
+BASE_URL = f'{API_URL}/repos/{OWNER}/{REPO}'
+BRANCH = 'master'
+
+
+def get_projects_path(pr_number, headers):
+  """Gets the current project path."""
+  response = requests.get(f'{BASE_URL}/pulls/{pr_number}/files',
+                          headers=headers)
+
+  projects_path = set()
+  for file in response.json():
+    file_path = file['filename']
+    dir_path = os.path.dirname(file_path)
+    if 'projects' in dir_path:
+      projects_path.add(dir_path)
+  return list(projects_path)
+
+
+def get_author_email(pr_number, headers):
+  """Retrieves the author's email address for a pull request,
+  including non-public emails."""
+  commits_response = requests.get(f'{BASE_URL}/pulls/{pr_number}/commits',
+                                  headers=headers)
+  if not commits_response.ok:
+    return None
+  email = commits_response.json()[0]['commit']['author']['email']
+  return email
+
+
+def get_project_yaml(project_path, headers):
+  """Gets the project yaml file."""
+  contents_url = f'{BASE_URL}/contents/{project_path}/project.yaml'
+  return get_yaml_file_content(contents_url, headers)
+
+
+def get_yaml_file_content(contents_url, headers):
+  """Gets yaml file content."""
+  response = requests.get(contents_url, headers=headers)
+  if not response.ok:
+    return {}
+  content = base64.b64decode(response.json()['content']).decode('UTF-8')
+  return yaml.safe_load(content)
+
+
+def get_integrated_project_info(pr_number, headers):
+  """Gets the new integrated project."""
+  response = requests.get(f'{BASE_URL}/pulls/{pr_number}/files',
+                          headers=headers)
+
+  for file in response.json():
+    file_path = file['filename']
+    if 'project.yaml' in file_path:
+      return get_yaml_file_content(file['contents_url'], headers)
+
+  return None
+
+
+def get_criticality_score(repo_url):
+  """Gets the criticality score of the project."""
+  report = subprocess.run([
+      '/home/runner/go/bin/criticality_score', '--format', 'json',
```

Comment:
nit: Can you put this in a CONSTANT global in this file? Alternatively, if /home/runner/go/bin is in PATH, use shutil.which to find the right path.